primitives/runtime/src/offchain/storage.rs

```diff
@@ -28,6 +28,35 @@ pub struct StorageValueRef<'a> {
 	kind: StorageKind,
 }
 
+/// Result of asking for a particular key from the store.
+#[derive(Debug, PartialEq, Eq)]
+pub enum StorageEntry<T> {
+	/// Successfully retrieved the stored value.
+	Occupied(T),
```

Comment:
We could get rid of the `Occupied(T)` entry, which makes this into `enum StorageEntryError`. Then later on we can return `Result<T, StorageEntryError>`, which enables some useful patterns.

Reply:
Ok, I'm good with Vacant being a type of error.